ArticleTemplates/assets/js/modules/messenger.js

```diff
@@ -0,0 +1,198 @@
+/* global Promise */
```

Comment:
I guess there are two options:
- polyfill.io: only clients with no support for Promise will get something back. The apparatus that Alex and Simon had to put in place to make this work is quite intricate, I don't know if it is a good idea.
- a bundled polyfill: works everywhere although affects all clients regardless of their level of support.